playbook-website/app/components/playbook/pb_docs/kit_example.rb

```diff
@@ -48,8 +48,47 @@ def tsx_source
         read_kit_file("", "_#{example_key}.tsx")
       end
 
+      def available_props
+        target_kit_path = ::Playbook.kit_path(kit, "", "_#{example_key}.tsx")
+        if File.exist?(cached_kit_target)
+          puts "AvailableProps: Using cached JSON for #{target_kit_path}"
+          return cached_kit_json
+        end
+
+        exec_in_fork do
+          `node scripts/react-docgen.mjs #{target_kit_path}`
+        end
+      end
+
     private
 
+      def exec_in_fork
```

Comment:
This creates a fork for the work to be done is parallel to any other requests on the server, hence non-blocking.